hushline/forms.py

```diff
@@ -25,5 +31,15 @@ def __call__(self, form: Form, field: Field) -> None:
             raise ValidationError(self.message)
 
 
+def is_valid_password_swap(
+    *, check_password: Callable[[str], bool], old_password: str, new_password: str
+) -> bool:
+    # since the passwords can be of different lengths, the equality test must occur *iif*
+    # the correctness test passes, since timing differences leak length information
+    if check_password(old_password):
+        return not bytes_are_equal(old_password.encode(), new_password.encode())
+    raise WrongPassword
```

Comment:
Raising an error isn't really consistent with the existing code style. I am open to refactoring the flow control to be error-based, but I'd prefer that as a separate PR.

Reply:
I also removed this pattern in the new PR #635